genesyscloud/team/genesyscloud_team_proxy.go

```diff
@@ -0,0 +1,174 @@
+func getTeamProxy(clientConfig *platformclientv2.Configuration) *teamProxy {
+	if internalProxy == nil {
+		internalProxy = newTeamProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// createTeam creates a Genesys Cloud team
+func (p *teamProxy) createTeam(ctx context.Context, team *platformclientv2.Team) (*platformclientv2.Team, error) {
+	return p.createTeamAttr(ctx, p, team)
+}
+
+// getTeam retrieves all Genesys Cloud team
+func (p *teamProxy) getAllTeam(ctx context.Context) (*[]platformclientv2.Team, error) {
+	return p.getAllTeamAttr(ctx, p)
+}
+
+// getTeamIdByName returns a single Genesys Cloud team by a name
+func (p *teamProxy) getTeamIdByName(ctx context.Context, name string) (id string, retryable bool, err error) {
+	return p.getTeamIdByNameAttr(ctx, p, name)
+}
+
+// getTeamById returns a single Genesys Cloud team by Id
+func (p *teamProxy) getTeamById(ctx context.Context, id string) (team *platformclientv2.Team, statusCode int, err error) {
+	return p.getTeamByIdAttr(ctx, p, id)
+}
+
+// updateTeam updates a Genesys Cloud team
+func (p *teamProxy) updateTeam(ctx context.Context, id string, team *platformclientv2.Team) (*platformclientv2.Team, error) {
+	return p.updateTeamAttr(ctx, p, id, team)
+}
+
+// deleteTeam deletes a Genesys Cloud team by Id
+func (p *teamProxy) deleteTeam(ctx context.Context, id string) (statusCode int, err error) {
+	return p.deleteTeamAttr(ctx, p, id)
+}
+
+// createTeamFn is an implementation function for creating a Genesys Cloud team
+func createTeamFn(ctx context.Context, p *teamProxy, team *platformclientv2.Team) (*platformclientv2.Team, error) {
+	team, _, err := p.teamsApi.PostTeams(*team)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create team: %s", err)
+	}
+
+	return team, nil
+}
+
+// getAllTeamFn is the implementation for retrieving all team in Genesys Cloud
+func getAllTeamFn(ctx context.Context, p *teamProxy) (*[]platformclientv2.Team, error) {
+	var allTeams []platformclientv2.Team
+	const pageSize = 100
+
+	teams, _, err := p.teamsApi.GetTeams(100, "", "", "", "")
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get team: %v", err)
+	}
+	if teams.Entities == nil || len(*teams.Entities) == 0 {
+		return &allTeams, nil
+	}
+	for _, team := range *teams.Entities {
+
+		allTeams = append(allTeams, team)
+	}
+
+	return &allTeams, nil
+}
+
+// getTeamIdByNameFn is an implementation of the function to get a Genesys Cloud team by name
+func getTeamIdByNameFn(ctx context.Context, p *teamProxy, name string) (id string, retryable bool, err error) {
+	teams, err := getAllTeamFn(ctx, p)
+	if err != nil {
+		return "", false, err
+	}
+
+	if len(*teams) == 0 {
+		return "", true, fmt.Errorf("No team found with name %s", name)
+	}
+
+	for _, team := range *teams {
+		if *team.Name == name {
+			log.Printf("Retrieved the team id %s by name %s", *team.Id, name)
+			return *team.Id, false, nil
+		}
+	}
+
+	return "", false, fmt.Errorf("Unable to find team with name %s", name)
```

Comment:
retryable should be true for this last return. Basically it should be retryable if everything went smoothly but the entity wasn't found

Reply:
done